src/libraries/System.Private.CoreLib/src/System/DateTimeOffset.Android.cs

```diff
@@ -29,52 +27,38 @@ public static DateTimeOffset Now
             {
                 DateTime utcDateTime = DateTime.UtcNow;
 
-                if (s_androidTZDataLoaded) // The background thread finished, the cache is loaded.
+                if (s_androidTZDataLoaded == 1) // The background thread finished, the cache is loaded.
+                {
                     return ToLocalTime(utcDateTime, true);
+                }
 
-                if (s_startNewBackgroundThread) // The cache isn't loaded and no background thread has been created
+                object? localDateTimeOffset = AppContext.GetData("System.TimeZoneInfo.LocalDateTimeOffset");
+                if (localDateTimeOffset == null) // If no offset property provided through monovm app context, default
                 {
-                    lock (s_localUtcOffsetLock)
-                    {
-                        // Now may be called multiple times before a cache is loaded and a background thread is running,
-                        // once the lock is available, check for a cache and background thread.
-                        if (s_androidTZDataLoaded)
-                            return ToLocalTime(utcDateTime, true);
+                    // no need to create the thread, load tzdata now
+                    s_androidTZDataLoaded = 1;
+                    return ToLocalTime(utcDateTime, true);
+                }
 
-                        if (s_loadAndroidTZData == null)
+                // The cache isn't loaded yet.
+                if (Interlocked.CompareExchange(ref s_androidTZDataLoaded, 0, -1) == -1)
+                {
+                    new Thread(() =>
+                    {
+                        try
                         {
-                            s_loadAndroidTZData = new Thread(() => {
-                                // Delay the background thread to avoid impacting startup, if it still coincides after 1s, startup is already perceived as slow
-                                Thread.Sleep(1000);
-
-                                _ = TimeZoneInfo.Local; // Load AndroidTZData
-                                s_androidTZDataLoaded = true;
+                            // Delay the background thread to avoid impacting startup, if it still coincides after 1s, startup is already perceived as slow
+                            Thread.Sleep(1000);
```

Comment:
could move the sleep to before creating the thread - ie delay even creating and starting the thread?